core/src/com/unciv/ui/multiplayer/FriendList.kt

```diff
@@ -0,0 +1,85 @@
+package com.unciv.ui.multiplayer
+
+import com.unciv.UncivGame
+
+class FriendList {
+    private val settings = UncivGame.Current.settings
+    var friendList = settings.multiplayer.friendList
+
+    enum class ErrorType {
+        NOERROR,
+        NAME,
+        ID,
+        NONAME,
+        NOID,
+        YOURSELF,
+        ALREADYINLIST;
+    }
+
+    data class Friend(val name: String, val playerID: String) {
+        constructor() : this("", "")
+    }
+
+    fun addNewFriend(friendName: String, playerID: String): ErrorType {
+        for(index in friendList.indices){
+            if (friendList[index].name == friendName) {
+                return ErrorType.NAME
+            } else if (friendList[index].playerID == playerID) {
+                return ErrorType.ID
+            }
+        }
+        if (friendName == "") {
+            return ErrorType.NONAME
+        } else if (playerID == "") {
+            return ErrorType.NOID
+        } else if (playerID == UncivGame.Current.settings.multiplayer.userId) {
+            return ErrorType.YOURSELF
+        }
+        friendList.add(Friend(friendName, playerID))
+        settings.save()
+        return ErrorType.NOERROR
+    }
+
+    fun editFriend(friend: Friend, name: String, playerID: String) {
+        friendList.remove(friend)
+        val editedFriend = Friend(name,playerID)
+        friendList.add(editedFriend)
+        settings.save()
+    }
+
+    fun deleteFriend(friend: Friend) {
+        friendList.remove(friend)
+        settings.save()
+    }
+
+    fun getFriendsList(): MutableList<Friend> {
```

Comment:
Your reference field is public, so any client could just use the field instead of the accessor... Like PlayerPickerTable does. The pattern is - you use accessors to control access, so the field would be private. If you're fine with a public field, no accessor needed? Also, FYI, `fun getFriendsList() = friendList` would compile exactly the same, the only difference being less options when debugging, and in very complex cases the type inference engine can run around in circles, so an explicit type here or there might help. Or it helps the reader... Here, it's simple. Mind - I'm not saying change that, just pointing out tidbits. Ah, another tidbit - the accessor pattern is ingrained in Java, and kotlin uses JVM - so if there weren't the one letter 's' difference, this would get you a complaint that the field getter and this accessor have the same JVM name... Which would disappear the instant you make the field private. Or use `@ JvmName`... Curious, ain't it?